docs/spec/annotations.rst

```diff
@@ -53,42 +53,151 @@ decorators ``@property``, ``@staticmethod`` and ``@classmethod``.
 
 .. _valid-types:
 
-Valid type expression forms
----------------------------
-
-Type hints may be built-in classes (including those defined in
-standard library or third-party extension modules), abstract base
-classes, types available in the ``types`` module, and user-defined
-classes (including those defined in the standard library or
-third-party modules).
-
-While annotations are normally the best format for type hints,
-there are times when it is more appropriate to represent them
-by a special comment, or in a separately distributed stub
-file.  (See below for examples.)
+Type and annotation expressions
+-------------------------------
+
+The terms *type expression* and *annotation expression* denote specific
+subsets of Python expressions that are used in the type system.
```

Comment:
Would it make sense to clarify that all type expressions are annotation expressions, but some annotation expressions aren't type expressions? (E.g. `ClassDef[...]`.)

Reply:
Added